moldock/preparation_for_docking.py

```diff
@@ -165,20 +170,44 @@ def fix_pdbqt(pdbqt_block):
     return '\n'.join(pdbqt_fixed)
 
 
+def assign_bonds_from_template(template_mol, mol):
+    # explicit hydrogends are removed from carbon atoms (chiral hydrogens) to match pdbqt mol,
+    # e.g. [NH3+][C@H](C)C(=O)[O-]
+    template_mol = Chem.AddHs(template_mol, explicitOnly=True,
+                              onlyOnAtoms=[a.GetIdx() for a in template_mol.GetAtoms() if
+                                           a.GetAtomicNum() != 6])
+    mol = AllChem.AssignBondOrdersFromTemplate(template_mol, mol)
+    Chem.SanitizeMol(mol)
+    Chem.AssignStereochemistry(mol, cleanIt=True, force=True, flagPossibleStereoCenters=True)
+    return mol
+
+
+def boron_reduction(mol_B, mol):
+    idx_boron = [idx for idx, atom in enumerate(mol_B.GetAtoms()) if atom.GetAtomicNum() == 5]
+    for id_ in idx_boron:
+        mol_B.GetAtomWithIdx(id_).SetAtomicNum(6)
+    mol = assign_bonds_from_template(mol_B, mol)
+    idx = mol.GetSubstructMatches(mol_B)
+    mol_idx_boron = [tuple(sorted(ids[i] for i in idx_boron)) for ids in idx]
+    mol_idx_boron = list(set(mol_idx_boron))
+    if len(mol_idx_boron) == 1:
+        for i in mol_idx_boron[0]:
+            mol.GetAtomWithIdx(i).SetAtomicNum(5)
+    else:
+        print('different mappings was detected. The structure cannot be recostructed automatically.')
```

Comment:
1. replace `print` with `sys.stderr.write`
2. return None, if the check failed, to make behavior of the program consistent and predictable (we will intercept and treat this exception later)